pkg/operator/ceph/object/spec.go

```diff
@@ -76,6 +76,19 @@ func (c *clusterConfig) createDeployment(rgwConfig *rgwConfig) (*apps.Deployment
 			MaxSurge:       &intstr.IntOrString{IntVal: int32(0)},
 		}
 	}
+	hpas, err := c.context.Clientset.AutoscalingV1().HorizontalPodAutoscalers(c.store.Namespace).List(c.clusterInfo.Context, metav1.ListOptions{LabelSelector: c.storeLabelSelector()})
```

Comment:
I was not able to find `field selector` for HPA, anything which I used returns an error saying only `name` and `namespace` can be used as field selector. So I recommend users add labels for the ScaledObject Resource, this will passed to HPA. And checking `HPA` than `Scaled Object` since ai-resource reality is available for HPA in k8s go SDK client.